### ISSUE TITLE
sof modules unloading failure on GLK

### ISSUE THREAD

**Issue by keqiaozhang** [thesofproject/linux#680]
This issue happens when unloading the modules. dai links are missing in sof_link_unload. 

How to reproduce:
1. remove the modules
```
rmmod sof_pci_dev
rmmod snd_sof_intel_hda_common
rmmod snd_sof_intel_hda
rmmod snd_sof_xtensa_dsp
rmmod snd_soc_acpi_intel_match
rmmod snd_sof
rmmod snd_soc_sst_bxt_da7219_max98357a
rmmod snd_soc_da7219
rmmod snd_soc_hdac_hdmi
rmmod snd_soc_dmic
rmmod snd_soc_max98357a
rmmod snd_soc_acpi
rmmod snd_hda_ext_core
rmmod snd_hda_core
```
2. check the dmesg.

dmesg:
```
[   73.909837] sof-audio-pci 0000:00:0e.0: error: failed to find dai HDMI3 in sof_link_unload
[   73.909848] sof-audio-pci 0000:00:0e.0: error: failed to find dai HDMI2 in sof_link_unload
[   73.909853] sof-audio-pci 0000:00:0e.0: error: failed to find dai HDMI1 in sof_link_unload
[   73.909858] sof-audio-pci 0000:00:0e.0: error: failed to find dai DMIC01 in sof_link_unload
[   73.909861] sof-audio-pci 0000:00:0e.0: error: failed to find dai Headset in sof_link_unload
[   73.909866] sof-audio-pci 0000:00:0e.0: error: failed to find dai Speakers in sof_link_unload
```
Test recipe:
sof-master: [82d29e0](https://github.com/thesofproject/sof/commit/82d29e099cd75c413ed7431e3e27d145d7a127b6)
kernel: topic/cros-sof-v4.14-rebase-b61733c
[dmesg.log](https://github.com/thesofproject/linux/files/2909060/dmesg.log)

**Comment by plbossart** [comment]
@bardliao can you check if there is one of your patches missing on this glk branch?

**Comment by bardliao** [comment]
@plbossart Yes, [ASoC: SOF: topology: check no_pcm in sof_link_unload](https://github.com/thesofproject/linux/commit/09e0018857379a98e726c185d9aee4c964eaacc9#diff-5e8a1eb50aedd2de9724071bfa7b6ebb) is missing. Thanks.

**Comment by keqiaozhang** [comment]
@bardliao 
Confirmed that above patch can fix this issue.
@ranj063  
Please help to backport this patch to cros branch.

**Comment by ranj063** [comment]
@keqiaozhang done!

**Comment by keqiaozhang** [comment]
Thanks @ranj063 .
This issue is fixed in cros-sof-v4.14-rebase: 164bdbb.
